File: fixtures/scale/php/large/src/Families/Columnar/ColumnarDescriptor.php
<?php

namespace CipherLab\Families\Columnar;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together ColumnarCipher and
// ColumnarVectors under the "transposition" suite.
class ColumnarDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'columnar';
    }

    public function suite(): string
    {
        return 'transposition';
    }

    public function cipher(): Cipher
    {
        return new ColumnarCipher();
    }

    public function vectors(): array
    {
        return ColumnarVectors::all();
    }
}

File: fixtures/scale/objc/large/Families/Columnar/ColumnarDescriptor.m
#import "ColumnarDescriptor.h"
#import "ColumnarCipher.h"
#import "ColumnarVectors.h"

@implementation ColumnarDescriptor

- (NSString *)family {
    return @"columnar";
}

- (NSString *)suite {
    return @"transposition";
}

- (id<CLCipher>)cipher {
    return [[ColumnarCipher alloc] init];
}

- (NSArray *)vectors {
    return [ColumnarVectors all];
}

@end
